app/models/upload_cleanup_job.rb

```diff
@@ -7,26 +7,27 @@
 
 class UploadCleanupJob < Struct.new(:study, :study_file, :retry_count)
   extend ErrorTracker
+  MAX_RETRIES = 2
 
   def perform
-    retries = retry_count + 1
+    self.retry_count += 1
```

Comment:
I don't see you initializing this to zero anywhere?   is this some ruby magic where nil gets auto-converted to zero?   That could probably use a comment (even if it is an established rubyism, it throws off folks who spend most of their time in other languages)

Reply:
It's set to 0 when the job is queued - see [here](https://github.com/broadinstitute/single_cell_portal_core/blob/jb-upload-cleanup-limit/app/models/study.rb#L1503), for instance.  But I agree we could add a `retry_count ||= 0` at the top to ensure that.